### PR TITLE
Working with DTO first draft

### PR DESCRIPTION
**Pull request by irinaponomarevahaulmont**
Please review.

### CODE REVIEW

File: content/modules/studio/pages/working-with-dtos.adoc [hunk 1]
```diff
@@ -0,0 +1,377 @@
...
+
+[source, java]
+/**
+ * DTO for {@link com.company.citycountrydemo.dto.Institution}
+ */
+
+[[generate-from-classes]]
+== Generating DTOs from any Java/Kotlin classes
+
+Nowadays, the DTO pattern is widely used in software development. It is not only used with JPA entities, but also with regular POJO classes. With Studio, you are not restricted to using DTOs with just JPA entities. You can create DTOs from any Java or Kotlin class, which gives you more flexibility and control over your code. For example, you can use Studio with MongoDB documents.
```
Comment (alexbudarov) [inline content/modules/studio/pages/working-with-dtos.adoc:203]: First two sentences look like a broad introduction. We don't need them in the middle of this chapter.

File: content/modules/studio/pages/working-with-dtos.adoc [hunk 2]
```diff
@@ -0,0 +1,377 @@
...
+
+We do it in the usual way, and for the `play` attribute we select the *_Existing Class_* option.
+
+image::dto-mapping-methods1.png[align=center]
+
+We also create a new MapStruct interface for it and name it simply `Mapper`, since it will be used for all entities.
+
+image::dto-mapping-methods2.png[align=center]
+
+Once we click OK, the new classes CharacterDto and Mapper will be generated. But now we need to add more methods to it - the methods to convert Play and Author entities to DTO and back.
```
Comment (alexbudarov) [inline content/modules/studio/pages/working-with-dtos.adoc:236]: styling as code

File: content/modules/studio/pages/working-with-dtos.adoc [hunk 3]
```diff
@@ -0,0 +1,377 @@
...
+
+//TODO write up the section once the functionality is added to the plugin.
+
+[[keeping-in-sync]]
+== Keeping DTO in sync with its JPA entity
+
+[[refactoring-attributes]]
+=== Refactoring attributes
+
+DTOs are commonly used at the API controller level to define only the fields required by the client. This is why DTOs nearly copy the structure of their entities. There are popular frameworks to map entities to DTOs and vice versa: *_MapStruct_* and *_ModelMapper_*. They auto-map namesake properties. However, changing the property name in an entity often leads to a corrupted mapping logic. Studio helps developers refactor entity properties along with their related fields in DTOs:
```
Comment (alexbudarov) [inline content/modules/studio/pages/working-with-dtos.adoc:296]: don't mention ModelMapper.
probably, should reformulate the sentence, like:
with Amplicode, you can use MapStruct library to map entities to DTOs and vice versa.

File: content/modules/studio/pages/working-with-dtos.adoc [hunk 4]
```diff
@@ -0,0 +1,377 @@
...
+
+For example, if we use the *_Refactor_*->*_Rename_* option to rename the `name` attribute of the `City` class into `cityName`, the following window will appear:
+
+image::dto-renaming-attribute.png[align=center]
+
+It will suggest all the necessary changes in the names of methods, components and parameters including those located n the corresponding DTO class. We just need to decide which items in this list should be renamed, uncheck the rest and click OK.
```
Comment (alexbudarov) [inline content/modules/studio/pages/working-with-dtos.adoc:302]: located n

File: content/modules/studio/images/dto-navigation.png
Comment (alexbudarov) [inline content/modules/studio/images/dto-navigation.png:1]: Contents on the screenshot is too small.
I suggest to not show project tree, and resize a bit the IDE window to save some horizontal space.

**Review by alexbudarov — CHANGES_REQUESTED** [review]

**Comment by irinaponomarevahaulmont** [comment]
I have applied the requested changes, please review.

File: content/modules/studio/pages/working-with-dtos.adoc [hunk 5]
```diff
@@ -0,0 +1,377 @@
...
+ */
+
+Studio will generate it automatically. The gutter icon will appear to help us navigate between the DTO and the entity. If the original entity has validation annotations, they will be copied into the DTO.
+
+[[entities-from-pojos]]
+== Generate Entities from POJOs
+
+Studio provides an *_Entity from POJO_* action that helps us to generate a JPA entity from any Java/Kotlin class. This feature may be helpful if you develop your application following the API-first approach: define DTOs for the API first and implement the data model later.
+
+In order to do this, use the *_JPA Entity from POJO_* option available both in the Studio drop-down menu and in the main menu of your IDE.
```
[the remark this answers is not in the document]
Comment (alexbudarov) [reply content/modules/studio/pages/working-with-dtos.adoc]: You have not changed two similar captions below, headings of screenshots.

[the remark this answers is not in the document]
Comment (alexbudarov) [reply content/modules/studio/pages/working-with-dtos.adoc]: >  in the Context menu in the project tree

context menu - can be written in lower case, it's a term in user interfaces

File: content/modules/studio/pages/working-with-dtos.adoc [hunk 6]
```diff
@@ -0,0 +1,388 @@
...
+
+[[generate-from-classes]]
+== Generating DTOs from any Java/Kotlin classes
+
+With Studio, you are not restricted to using DTOs with just JPA entities. You can create DTOs from any Java or Kotlin class, which gives you more flexibility and control over your code. For example, you can use Studio with MongoDB documents.
+
+[[mapstruct-support]]
+== MapStruct Support
+
+https://mapstruct.org/[*_MapStruct_*] is a code generator that greatly simplifies the implementation of mappings. The *_Mapper class_* field appears in the *_New DTO_* window if your project contains the corresponding dependency. You can select an existing Mapper or create a new one.
```
Comment (alexbudarov) [inline content/modules/studio/pages/working-with-dtos.adoc:217]: Change this link to a direct link to documentation:
https://mapstruct.org/documentation/1.4/reference/html/

And let's make this link a variable, to be defined outside of .adoc contents. So I can update it later easily when we will update platform's libraries.
As example, search for "main-site" in amplicode-docs, how it is defined in .yml files and how it is used in the content.

**Review by alexbudarov — CHANGES_REQUESTED** [review]
few changes left

**Comment by irinaponomarevahaulmont** [comment]
Requested changes addressed, please review.

**Pull request merged**